### DISCUSSION TITLE
Add a disposing function callback to put(), lazyPut(), etc.

### DISCUSSION THREAD

**Discussion by OliverRhyme** [jonataslaw/getx#1292] [Ideas]
**Is your feature request related to a problem? Please describe.**
I am not able to properly clear a getx dependency (I.E moor database) as the object requires a cleanup method to be called for it to be properly disposed.

**Describe the solution you'd like**
A disposing callback function that is called just before it is "deleted" by the getx. It should pass the current object to be deleted as the function parameter and should/can be an async function. Pretty much like get_it package disposing callbacks on thier register methods.

**Describe alternatives you've considered**
I've tried adding the GetLifecycleBase as a mixin to the class but it just makes the code "spaghetti" as all I want is to call a method on the object just before it is deleted by GetX.

**Comment by jonataslaw** [comment]
I don't think this feature is of any use, onClose already does exactly that job.
Moving to discussions, if there is community movement in this direction to support or add this resource, it will be added.

**Reply by OliverRhyme** [reply]
Is onClose only available for the controllers? Also calling Get.find on the onClose for the corresponding controller does not return the object requested (disposed first) before having the chance to call disposing function on the object.

**Comment by OliverRhyme** [comment]
```
void main() {
  runApp(
    Provider<MyDatabase>(
      create: (context) => MyDatabase(),
      child: MyFlutterApp(),
      dispose: (context, db) => db.close(),
   ),
  );
}
```
or how can I implement this kind of behavior in getx?
